src/d_main.c

```diff
@@ -2127,30 +2150,49 @@ void D_DoomMain(void)
   // @category game
   // @arg <skill>
   // @vanilla
+  // @help
   //
-  // Set the game skill, 1-5 (1: easiest, 5: hardest).  A skill of
-  // 0 disables all monsters.
+  // Set the game skill, 1-5 (1: easiest, 5: hardest). A skill of 0 disables all
+  // monsters.
   //
 
   if ((p = M_CheckParm ("-skill")) && p < myargc-1)
-    {
-      startskill = myargv[p+1][0]-'1';
-      autostart = true;
-    }
+   {
+     startskill = M_ParmArgToInt(p);
+     startskill--;
+     if (startskill >= sk_none && startskill <= sk_nightmare)
+      {
+        autostart = true;
+      }
+     else
+      {
+        I_Error("Invalid parameter '%s' for -skill, valid values are 1-5 "
+                "(1: easiest, 5: hardest). "
+                "A skill of 0 disables all monsters.", myargv[p+1]);
+      }
+   }
 
   //!
   // @category game
   // @arg <n>
   // @vanilla
   //
-  // Start playing on episode n (1-4)
+  // Start playing on episode n (0-99)
   //
 
   if ((p = M_CheckParm ("-episode")) && p < myargc-1)
     {
-      startepisode = myargv[p+1][0]-'0';
-      startmap = 1;
-      autostart = true;
+      startepisode = M_ParmArgToInt(p);
+      if (startepisode >= 0 && startepisode <= 99)
```

Comment:
We are still unconditionally restricting episode to be > 0:

https://github.com/fabiangreffrath/woof/blob/d8a068c97b3e98b918d31360ad84c5b22f43c249/src/g_game.c#L2982-L2983